app/common/test/services/testOrderResultService.js

```diff
@@ -70,18 +71,43 @@
 
     function buildTestResultResource(provider, testResult, item) {
       return {
-        encounterRequest: {uuid: testResult.encounterRequest.uuid},
+        encounterRequest: { uuid: testResult.encounterRequest.uuid },
         dateCreation: moment().format('YYYY-MM-DD'),
-        provider: {uuid: provider.uuid},
+        provider: { uuid: provider.uuid },
         items: [
           {
-            "testOrder": {"type": item.testOrder.type, "uuid": item.testOrder.uuid},
+            "testOrder": { "type": item.testOrder.type, "uuid": item.testOrder.uuid },
             "value": "" + item.value
           }
         ]
       };
     }
 
+    function getTestOrderConsolidateResult(testRequestUuid, testOrderUuid) {
+      return getTestOrderResult(testRequestUuid)
+        .then(function (testRequest) {
+          var testOrders = testRequest.items;
+          var testOrderWithResult = testOrders.find(function (testOrder) { return testOrder.uuid === testOrderUuid; });
+          return conceptService.getConceptByTestOrder(testOrderWithResult.uuid, 'custom:(uuid,display,answers,datatype,units)')
+            .then(function (concept) {
+              return $q(function (resolve) {
```

Comment:
@edrisse Its not really necessary to return a promise from ```then```, you can just return the result.

https://docs.angularjs.org/api/ng/service/$q#chaining-promises